module/workflow/src/Infrastructure/Query/ProductWorkflowQuery.php

```diff
@@ -83,4 +83,35 @@ public function getQuery(
 
         return $result;
     }
+
+    /**
+     * @return array
+     *
+     * @throws \ReflectionException
+     */
+    public function getAvailableStatuses(
+        AbstractProduct $product,
+        AbstractWorkflow $workflow,
+        Language $productLanguage
+    ): array {
+        $code = new AttributeCode(StatusSystemAttribute::CODE);
+        $result = [];
+        if ($product->hasAttribute($code)) {
+            $attributeId = $this->attributeQuery->findAttributeIdByCode($code);
+            Assert::notNull($attributeId, sprintf('attribute %s not exists', $attributeId->getValue()));
+            $value = $product->getAttribute($code)->getValue();
+            $statusId = new StatusId($value[$productLanguage->getCode()]);
+
+            $transitions = $workflow->getTransitionsFromStatus($statusId);
+            foreach ($transitions as $transition) {
+                if ($this->service->available($transition, $product, $productLanguage)) {
+                    $fromStatus = $this->statusRepository->load($transition->getTo());
```

Comment:
Wouldn't it suffice to add the id to the result? Loading status seems excessive since it's fetched from the workflow itself